### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by marandaneto**
#skip-changelog

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-dart-plugin/issues/95

## :green_heart: How did you test it?

CI

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@vaind I gave it a quick try but there are more problems, wondering if you have upgraded the CL somewhere else and if it also failed since the scripts are pretty much the same?

File: integration-test/assets/release.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "deployCount": 0,
     "dateCreated": "2022-01-01T12:00:00.000000Z",
     "lastEvent": null,
-    "version": "1.1.0",
+    "version": "project@1.1.0",
```
Comment (marandaneto) [inline integration-test/assets/release.json:12]: There was a version mismatch here.

File: integration-test/scripts/test-server.py [hunk 2]
```diff
@@ -41,11 +41,11 @@ def do_GET(self):
                            '"chunkSize":8388608,"chunksPerRequest":64,"maxFileSize":2147483648,'
                            '"maxRequestSize":33554432,"concurrency":1,"hashAlgorithm":"sha1","compression":["gzip"],'
                            '"accept":["debug_files","release_files","pdbs","sources","bcsymbolmaps"]}')
-        elif self.isApi('/api/0/organizations/{}/repos/?cursor='.format(apiOrg)):
+        elif self.isApi('/api/0/organizations/{}/repos/'.format(apiOrg)):
             self.writeJSONFile("assets/repos.json")
         elif self.isApi('/api/0/organizations/{}/releases/{}@{}/previous-with-commits/'.format(apiOrg, appIdentifier, version)):
             self.writeJSON('{ }')
-        elif self.isApi('/api/0/projects/{}/{}/releases/{}/files/?cursor='.format(apiOrg, apiProject, version)):
+        elif self.isApi('/api/0/projects/{}/{}/releases/{}@{}/files/'.format(apiOrg, apiProject, appIdentifier, version)):
```
Comment (marandaneto) [inline integration-test/scripts/test-server.py:48]: version was missing

**Review by vaind — COMMENTED** [review]
The changes look reasonable to me. This version of the script you have here is, IIRC, actually taken from an @denrase's adaptation of the original script. Maybe he's aware of the changes.

FYI, the latest version I'm aware of (albeit not tested with the most recent CLI) is at https://github.com/getsentry/github-workflows/pull/54/files#diff-55e89cacc059af14ee09967d3f45e59bfc60b1c797319271e8e1b135830ce978 where I started extracting this whole thing to a workflow and used it in the dotnet SDK. This isn't finished yet though but when it is, I'll check if it could be used here as well. In theory, it should be usable in all the repos although at the moment, the APIs that were handled by the script were based on a need-to-know basis, i.e. if the specific tool didn't send a request at the time the integration test was created, we didn't do API handling. Trying to handle all Sentry APIs would be too much work for little benefit.

**Comment by krystofwoldrich** [comment]
I did know about his PR, when I merged https://github.com/getsentry/sentry-dart-plugin/pull/103, but from what I see here the plugin works, it just the mocked server that is not working at the moment?

File: integration-test/scripts/test-server.py [hunk 3]
```diff
@@ -130,7 +130,9 @@ def requestBody(self):
         return None
 
     def isApi(self, api: str):
-        if self.path.strip('/') == api.strip('/'):
+        noQueryAPI = api.strip('?')[0]
+        noQueryPath = self.path.strip('?')[0]
```
[the remark this answers is not in the document]
Comment (denrase) [reply integration-test/scripts/test-server.py]: Switched `strip` to `split` here, think this is what was intended as matching with strip was broken.

**Comment by marandaneto** [comment]
@denrase 

> warning - test/cli_test.dart:15:19 - This API is available since SDK 2.15.0, but constraints '>=2.12.0 <3.0.0' don't guarantee it. Try updating the SDK constraints. - sdk_version_since
> warning - test/cli_test.dart:20:31 - This API is available since SDK 2.15.0, but constraints '>=2.12.0 <3.0.0' don't guarantee it. Try updating the SDK constraints. - sdk_version_since

**Pull request merged**